swarm/network/kademlia.go

```diff
@@ -82,13 +82,15 @@ func NewKadParams() *KadParams {
 
 // Kademlia is a table of live peers and a db of known peers (node records)
 type Kademlia struct {
-	lock       sync.RWMutex
-	*KadParams            // Kademlia configuration parameters
-	base       []byte     // immutable baseaddress of the table
-	addrs      *pot.Pot   // pots container for known peer addresses
-	conns      *pot.Pot   // pots container for live peer connections
-	depth      uint8      // stores the last current depth of saturation
-	depthC     chan uint8 // returned by DepthC function to signal depth change
+	lock             sync.RWMutex
+	*KadParams                // Kademlia configuration parameters
+	base             []byte   // immutable baseaddress of the table
+	addrs            *pot.Pot // pots container for known peer addresses
+	conns            *pot.Pot // pots container for live peer connections
+	depth            uint8    // stores the last current depth of saturation
+	nDepth           int      // stores the last neighbourhood depth
+	nDepthC          chan int // returned by DepthC function to signal neighbourhood depth change
+	addressBookSizeC chan int // returned by AddressBookSizeC function to signal peer count change
```

Comment:
call it `addrCountC`?

i know it was my idea but address book is not used elsewhere

Reply:
It is used here https://github.com/ethersphere/go-ethereum/pull/313/files#diff-524f169c33e854ca57a6e668ce319a9bR135 over AddressBookSizec function.